pkg/operator/controller/instance_controller.go

```diff
@@ -0,0 +1,742 @@
+	if err := r.setPVC(ctx, ControlPlaneName); err != nil {
+		return err
+	}
+
+	// Set services
+	if err := r.setService(ctx, ControlPlaneName, cpapi.ListenPort); err != nil {
+		return err
+	}
+	if err := r.setService(ctx, DataPlaneName, dpapi.ListenPort); err != nil {
+		return err
+	}
+
+	// Set deployments
+	if err := r.setControlplane(ctx); err != nil {
+		return err
+	}
+	if err := r.setDataplane(ctx); err != nil {
+		return err
+	}
+
+	// Set rules
+	if err := r.setRules(ctx, ControlPlaneName); err != nil {
+		return err
+	}
+
+	// Set External service
+	return r.setExternalService(ctx)
+}
+
+// setControlplane sets up the controlplane deployment.
+func (r *InstanceReconciler) setControlplane(ctx context.Context) error {
+	cpDeployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ControlPlaneName,
+			Namespace: r.clNamespace,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: int32Ptr(1),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"app": ControlPlaneName},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app": ControlPlaneName},
+				},
+				Spec: corev1.PodSpec{
+					Volumes: []corev1.Volume{
+						{
+							Name: "ca",
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: "cl-fabric",
+								},
+							},
+						},
+						{
+							Name: "tls",
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: ControlPlaneName,
+								},
+							},
+						},
+						{
+							Name: ControlPlaneName,
+							VolumeSource: corev1.VolumeSource{
+								PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+									ClaimName: ControlPlaneName,
+								},
+							},
+						},
+					},
+					Containers: []corev1.Container{
+						{
+							Name:            ControlPlaneName,
+							Image:           r.instance.Spec.ContainerRegistry + ControlPlaneName + ":" + r.instance.Spec.ImageTag,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Args:            []string{"--log-level", r.instance.Spec.LogLevel, "--platform", "k8s"},
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: cpapi.ListenPort,
+								},
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "ca",
+									MountPath: cpapp.CAFile,
+									SubPath:   "ca",
+									ReadOnly:  true,
+								},
+								{
+									Name:      "tls",
+									MountPath: cpapp.CertificateFile,
+									SubPath:   "cert",
+									ReadOnly:  true,
+								},
+								{
+									Name:      "tls",
+									MountPath: cpapp.KeyFile,
+									SubPath:   "key",
+									ReadOnly:  true,
+								},
+								{
+									Name:      ControlPlaneName,
+									MountPath: filepath.Dir(cpapp.StoreFile),
+								},
+							},
+							Env: []corev1.EnvVar{
+								{
+									Name: "CL-NAMESPACE",
```

Comment:
See earlier note on hardcoding the namespace name. It can be made available to the operator pods in the same manner as you provide it to the control plane.

Reply:
Part of the controller (@orozery )